cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/common/TrackedEmitter.java

```diff
@@ -31,11 +31,16 @@
   private final String emitMetricName;
 
   public TrackedEmitter(Emitter<T> delegate, StageMetrics stageMetrics, String emitMetricName) {
+
     this.delegate = delegate;
     this.stageMetrics = stageMetrics;
     this.emitMetricName = emitMetricName;
   }
 
+  public Emitter<T> getEmitter() {
```

Comment:
this type of thing destroys the abstraction and encourages bad behavior. It will quickly become impossible to tell when metrics are emitted and when they aren't. It seems like its only used when getting the previous stage? Will it be removed when that part is cleaned up?